gli/io.py

```diff
@@ -1451,3 +1470,871 @@ def save_task_node_classification(name,
+    :type test_neg: list/array of int or list of lists/2-d array of int
+    :param task_id: The task ID. This is needed when there are multiple tasks
+        of the same task type are defined on the dataset. Default: 1.
+    :type task_id: int
+    :param save_dir: The directory to save the task json and data files.
+        Default: ".".
+    :type save_dir: str
+
+    :raises ValueError: If `description` is not a string.
+    :raises ValueError: If `time` is not a string.
+    :raises ValueError: If `time` do not correspond to edge attributes.
+    :raises ValueError: If `train_time_window` is not a list.
+    :raises ValueError: If `val_time_window` is not a list.
+    :raises ValueError: If `test_time_window` is not a list.
+    :raises ValueError: If `feature` is not a list of strings.
+    :raises ValueError: If elements in `feature` do not correspond to
+        node/edge/graph attributes.
+
+    :return: The dictionary of the content in the task json file.
+    :rtype: dict
+
+    Example
+    -------
+    .. code-block:: python
+
+        # Save the task information.
+        save_task_time_dependent_link_prediction(
+            name="example_dataset",
+            description="A time dependent link prediction task for the example dataset.",
+            feature=["Node/DenseNodeFeature", "Node/SparseNodeFeature"],
+            time="Edge/EdgeYear")
+        # This function will save the task information into a json file named
+        # `task_node_classification_1.json` and one numpy data file storing the
+        # data splits, `train_set`, `val_set`, and `test_set`. The json file
+        # will look like the following.
+
+    .. code-block:: json
+
+        {
+            "description": "A time dependent link prediction task for the example dataset.",
+            "type": "TimeDependentLinkPrediction",
+            "feature": [
+                "Node/DenseNodeFeature",
+                "Node/SparseNodeFeature"
+            ],
+            "time": "Edge/EdgeYear",
+            "train_time_window": [
+                0,
+                0
+            ],
+            "val_time_window": [
+                0,
+                0
+            ],
+            "test_time_window": [
+                0,
+                0
+            ]
+        }
+    """  # noqa: E501,E262  #pylint: disable=line-too-long
+    # set default value
+    if train_time_window is None:
+        train_time_window = [0, 0]
+    if val_time_window is None:
+        val_time_window = [0, 0]
+    if test_time_window is None:
+        test_time_window = [0, 0]
+    # Check the input arguments.
+    assert isinstance(description, str), \
+        "`description` must be a string."
+    _check_feature(feature)
+    assert isinstance(time, str), \
+        "`time` must be a string."
+    assert time.startswith("Edge/"), \
+        "`time` must be an edge attribute."
+    assert isinstance(train_time_window, list), \
```

Comment:
Stronger tests for the time windows:

1. The time windows much be a list of two numbers
2. The time windows must not overlap